### PR TITLE
Added bounds checking for percentage in SimpleProportionalStrategy

### PR DESCRIPTION
**Pull request by ClifHouck**
With an exception and tests.

### CODE REVIEW

File: arsenal/strategy/simple_proportional_strategy.py [hunk 1]
```diff
@@ -111,8 +112,18 @@ def how_many_nodes_should_cache(nodes, percentage_to_cache):
...
+class InvalidPercentageError(exception.ArsenalException):
+    msg_fmt = ("An invalid percentage was specified. Percentages should be "
+               "less than or equal to 100, and greater than or equal to 0. "
+               "Got '%(percentage)f'.")
+
+
 class SimpleProportionalStrategy(object):
     def __init__(self, percentage_to_cache=25):
+        # Clamp the percentage to reasonable values.
+        if percentage_to_cache < 0 or percentage_to_cache > 100:
```
Comment (aweeks) [inline arsenal/strategy/simple_proportional_strategy.py]: We should use p ∈ [0, 1] to be consistent with style used in other stats libraries, and science.

**Comment by pcsforeducation** [comment]
+1 LGTM

File: arsenal/strategy/simple_proportional_strategy.py (hunk 1 above)
Comment (aweeks) [inline arsenal/strategy/simple_proportional_strategy.py]: I got a question from @pcsforeducation about this.  To be clear, I'm saying we should use probabilities, which are between 0 and 1inclusive, and avoid the informal percent (0 to 100 inclusive) notation.

Comment (ClifHouck) [inline arsenal/strategy/simple_proportional_strategy.py]: +1 for science, I will update it accordingly.

**Comment by aweeks** [comment]
Otherwise, LGTM

**Comment by ClifHouck** [comment]
Closing this as during addressing comments of this PR I discovered other issues that I've since addressed in a new PR, along with issues raised in the review of this PR.

**Pull request closed without merging**